### DISCUSSION TITLE
Additional Track display options

### DISCUSSION THREAD

**Discussion by SteveDaulton** [audacity/audacity#1686] [Ideas & Feature Requests]
Currently we have options to display a track as:
1. Waveform (linear scale)
2. Waveform (log scale)
3. Spectrogram (multiple options)

Each is a visualization of the underlying audio data, and provides information  to the user about the audio.

Although the waveform view may provide a clue about spacial (left / right) qualities, it is often difficult or impossible to glean much idea.

Displaying an audio track as Mid/Side would provide a much improved view of the stereo spread an relative phase, much like an audio goniometer. Unlike a goniometer, a Mid/Side display would show the spacial aspects for the entire track, rather than just the immediate playback position.

Mid/Side display could become  more useful with additional tools - such as a "Channel Mixer" that can operate either in Left/Right mode, or Mid/Side mode.

**Comment by LWinterberg** [comment]
I love this idea. The conversion is super simple (XY->MS: `M=L+R, S=L-R` with some scaling, MS->XY: `L=M+S, R=M-S`) 

The only concerns I'd have is that we'd need to explain the difference of MS/XY to users. But we probably should explain in design that top channel = L anyway, because this convention isn't exactly obvious - and while we're at it, we might as well use whatever system we come up to show that to explain MS, too. 

Also, Split stereo track in M/S view might be a bit weird.

**Reply by SteveDaulton** [reply]
> MS->XY: `L=M+S, R=M-S`)

The correct scaling is: `L=(M+S)/2, R=(M-S)/2`, which then makes the round trip L/R -> M/S -> L/R produce the same out as went in.

The M/S display should probably default to a vertical zoom level of +/- 2.0. For *normal* music, this is likely to be more than necessary, but ensures that the waveform fits.


![Tracks000](https://user-images.githubusercontent.com/11681097/133923785-f95446ab-fa19-462d-9a8e-482b289f052d.png)

One thing that may look strange is if "Show clipping" is enabled. If the M/S view is just a "view" of unaltered L/R audio data, then it may not be clipped  data beyond +/- 1, but if exported a M/S then it will be clipped at +/- 1.

> Also, Split stereo track in M/S view might be a bit weird.

Could be useful to convert the underlying data to M/S when splitting a M/S view. The "traditional" method for vocal removal is `L-R`, which is the side channel of M/S.
It would also allow Mid channels and Side channels to  be mixed separately when working  with multi-track projects.

The potential confusion, which needs some thought to display clearly, is whether it is just a "view" of the unchanged L/R data (so still plays as L/R stereo), or whether it is *converted* to M/S (which would  play the Mid channel on the  left and the Side channel on the right). This point is essentially the reason that I posted as a "Discussion" rather than a feature request :)

**Reply by SteveDaulton** [reply]
If anyone wants to play with channel mixing, there's a handy little plug-in here: https://wiki.audacityteam.org/wiki/Nyquist_Effect_Plug-ins#Channel_Mixer

**Comment by LWinterberg** [comment]
from the todo.txt: An average + difference display may be nice in addition to M/S

**Comment by LWinterberg** [comment]
2 more ideas have been brought to my attention: 

* A collapsed waveform that takes stereo (or one day also 5.1/7.1 etc) and collapses into a single line. The benefit of this is when you don't particularly care about the wave data but just want to assemble various sounds into a song, you don't have to deal with the additional complexity and space requirements of seeing two waveforms for one sound. 
* A 3-band waveform, which instead of using two colors for RMS/Peak would use 3 colors for peak low/mid/high. Useful for aligning to a kick drum in an otherwise heavily compressed track, also would be a simplified spectrogram.
